src/Auth.sol

```diff
@@ -1,16 +1,19 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity ^0.8.20;
 
+import { SimpleAuth } from "./SimpleAuth.sol";
+
 /// @title Auth
 /// @author Anna Carroll <https://github.com/anna-carroll/3074>
-abstract contract Auth {
+abstract contract Auth is SimpleAuth {
     /// @notice magic byte to disambiguate EIP-3074 signature payloads
     uint8 constant MAGIC = 0x04;
```

Comment:
Maybe move this down to `SimpleAuth`? `MAGIC` + `getDigest` here seems a bit low-level

Reply:
i was wondering whether to do this, sgtm :) https://github.com/clabby/eip-3074-foundry/pull/8/commits/f33d7d11506159f759ab170bf40b605833a305fb